cassandra/cassandra_test.go

```diff
@@ -0,0 +1,356 @@
+	time.Sleep(4 * time.Second)
+
+	// Default TTL key should be gone, no TTL key should remain
+	value, err = store.Get("key_default_ttl")
+	require.NoError(t, err)
+	require.Nil(t, value, "Key with default TTL should have expired")
+
+	value, err = store.Get("key_no_ttl")
+	require.NoError(t, err)
+	require.Equal(t, []byte("value3"), value, "Key with no TTL should still exist")
+}
+
+// / testReset tests the Reset method.
+func testReset(t *testing.T, connectionURL string, store *Storage) {
+
+	// Set some keys
+	err := store.Set("key1", []byte("value1"), 0)
+	require.NoError(t, err)
+
+	err = store.Set("key2", []byte("value2"), 0)
+	require.NoError(t, err)
+
+	// Verify keys exist
+	value, err := store.Get("key1")
+	require.NoError(t, err)
+	require.Equal(t, []byte("value1"), value)
+
+	// Reset storage
+	err = store.Reset()
+	require.NoError(t, err)
+
+	// Verify keys are gone
+	value, err = store.Get("key1")
+	require.NoError(t, err)
+	require.Nil(t, value, "Key should be deleted after reset")
+
+	value, err = store.Get("key2")
+	require.NoError(t, err)
+	require.Nil(t, value, "Key should be deleted after reset")
+
+	// Create new storage with Reset flag
+	store = New(Config{
+		Hosts:    []string{connectionURL},
+		Keyspace: "test_reset",
+		Table:    "test_kv",
+		Reset:    true,
+	})
+	defer store.Close()
+
+	// Set a key
+	err = store.Set("key3", []byte("value3"), 0)
+	require.NoError(t, err)
+
+	// Verify key exists
+	value, err = store.Get("key3")
+	require.NoError(t, err)
+	require.Equal(t, []byte("value3"), value)
+}
+
+// testConcurrentAccess tests concurrent access to the storage.
+func testConcurrentAccess(t *testing.T, store *Storage) {
+
+	// Number of goroutines
+	const concurrentOps = 10
+	done := make(chan bool, concurrentOps)
+
+	// Run concurrent operations
+	for i := 0; i < concurrentOps; i++ {
+		go func(id int) {
+			// Set key
+			key := fmt.Sprintf("key%d", id)
+			value := []byte(fmt.Sprintf("value%d", id))
+			err := store.Set(key, value, 0)
+			require.NoError(t, err)
+
+			// Get key
+			retrievedValue, err := store.Get(key)
+			require.NoError(t, err)
+			require.Equal(t, value, retrievedValue)
+
+			// Delete key
+			err = store.Delete(key)
+			require.NoError(t, err)
+
+			// Verify deletion
+			retrievedValue, err = store.Get(key)
+			require.NoError(t, err)
+			require.Nil(t, retrievedValue)
+
+			done <- true
+		}(i)
+	}
+
+	// Wait for all goroutines to complete
+	for i := 0; i < concurrentOps; i++ {
+		<-done
+	}
+}
+
+func Benchmark_Cassandra_Set(b *testing.B) {
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	connectionURL := newTestStore(b)
+
+	// Create new storage
+	store := New(Config{
+		Hosts:    []string{connectionURL},
+		Keyspace: "test_concurrent",
+		Table:    "test_kv",
+	})
+	defer store.Close()
+
+	var err error
+	for i := 0; i < b.N; i++ {
+		err = store.Set("john", []byte("doe"), 0)
+	}
+
+	require.NoError(b, err)
+}
+
+func Benchmark_Cassandra_Get(b *testing.B) {
+	b.ReportAllocs()
+	b.ResetTimer()
```

Comment:
suggestion: let's move this right before the for loop and the err declaration (see #1672)